src/Amazon.ElasticBeanstalk.Tools/Commands/DeployEnvironmentCommand.cs

```diff
@@ -415,7 +419,14 @@ private async Task<string> CreateEnvironment(string application, string environm
                     Value = loadBalancerType
                 });
             }
-           
+
+            // For new environments, disable IMDSv1 by default (unless explicitly enabled in additional options).
+            createRequest.OptionSettings.Add(new ConfigurationOptionSetting()
```

Comment:
Why does this need to be both the here and in `AddAdditionalOptions`? Seems redundant and you had to add code in `AddAdditionalOptions` to handle the fact that it was potentially added by the parent caller.

Reply:
I misunderstood the purpose here which was to force all new deployments to have disable to true. We don't want to do that because that could break users and we don't have a UI element in Visual Studio for users to toggle this.

@ashishdhingra is going to change the logic to remove this section and in the `AddAdditionalOptions` only set the disable if the user explicitly set the field. The VS toolkit can decide if they want to add a checkbox users can see and default the value to `true`.

Reply:
Removed the default disabling of IMDSv1 since it could break existing customers using legacy Visual Studio tooling. Also simplified code in `AddAdditionalOptions` to not check if existing setting is already added.